tests/components/mqtt/test_init.py

```diff
@@ -1292,8 +1292,9 @@ async def test_subscribe_same_topic(
 ) -> None:
     """Test subscribing to same topic twice and simulate retained messages.
 
-    When subscribing to the same topic again, SUBSCRIBE must be sent to the broker again
-    for it to resend any retained messages.
+    When subscribing to the same topic again, the last value sent to the topic
+    should be sent to the new subscriber (simulating exactly what happened with
+    the initial SUBSCRIBE call).
```

Comment:
This is not correct as the retained value would be send to the new subscriber, which is not necessarily the last payload received. But I agree this is better. Again, it is not what would happen when resubscribing.